### DISCUSSION TITLE
Susceptibility plot

### DISCUSSION THREAD

**Discussion by kg492** [Part-II-Computational-Physics/.github#67] [General]
Hello, I've been trying to debug the susceptibility plot to give the peak required, 

def evolve_and_computesusceptibility(lattice,betaJ,betaMuH,avg_times):
    M = []
    for t in range(avg_times[-1]+1):
        attempt_spin_flip(lattice,width,betaJ,beta
![susceptibility](https://user-images.githubusercontent.com/125891534/224093252-1d9346d5-b6d2-4048-bb15-5c50ff987401.png)
MuH)
        if t in avg_times:
            M.append(compute_magnetisation(lattice))
    susceptibility = (1/width**2) * np.var(M) 
    return susceptibility 

avg_times = [10000]
width = 40
lattice= init_lattice(width,0)
susceptibility = []
criticalcoupling = np.log(1+np.sqrt(2))/2
betaJs = [i*criticalcoupling for i in np.linspace(-5,5,200)]

for i in betaJs:
    susceptibility.append(evolve_and_computesusceptibility(lattice, i, 0 , avg_times))

plt.plot(betaJs, susceptibility)
plt.xlabel ('Coupling strength betaJ')
plt.ylabel ('Magnetic susceptibility ')
plt.show()

**Comment by cl804** [comment]
same issue here

**Reply by cl804** [reply]
It works for me now. Maybe you can change your average time have another try

**Comment by DLHvdH** [comment]
You have `avg_times = [10000]`. That means you will only get a single magnetisation value. Consider what this does to your computation of the variance.
